src/dhcp6.c

```
@@ -1790,7 +1790,12 @@ dhcp6_failconfirm(void *arg)
 	eloop_timeout_delete(ifp->ctx->eloop, dhcp6_sendconfirm, ifp);
 
 	/* RFC8415 18.2.3 says that prior addresses SHOULD be used on failure. */
-	dhcp6_fail(ifp, false);
+	/* but IPv6 Ready Logo DHCPv6 tests require it to be used, so provide
+	 * a way to do so */
+	if (ifp->options->options & DHCPCD_LASTLEASE)
```

I feel that this should be moved into `dhcp6_fail()` where the logic goes

```
if (drop) {
...
} else if (state->recv && ifp->options->options & DHCPCD_LASTLEASE) {
   dhcp6_bind()
} else if (state->new) {
 ...
}
```

That way, Prefix Delegation renew failure would be supported this way as well and that part of the RFC is also should so might be part of the tests.

What do you think? Also, can you test it? :)

---

Sure, I can make that change. I'm able to run the IPv6 Ready Logo tests, but my environment is kind of limited. I can only do the client tests, but not PD stuff.  And it takes up to 48 hours to run depending on the tests I pick, so it'll be a bit before I can get back to you.

---

That's fine. The lease code flow is exactly the same of what we want to do, so if your path works then the PD path will work just as well.

If either fail, we want to re-start the discovery process again as a server might not be authoritative and silently ignore us so we use our prior lease and then try and get a new one. This is the mirror of DHCP behavior so I think it will work well.

---

Looks good, but we now need to revert this section of code now the logic is in dhcp6_fail()

---

Sorry, I'm not sure what happened. The commit in my local branch has that already, not sure why it didn't show up in the PR. Let me just cancel this PR, clean it up and submit a new one. Sorry for the mess...